brave/brave-ios
# Fix #7794, #7709: Add Stripe on ramp, buy token tweaks

## Summary of Changes
- Adds support for Link by Stripe on ramp in buy token view. Stripe is only supported in the US.
- Selects USD currency as default for Buy
- Remove placeholder showing currency symbol to match desktop

This pull request fixes #7794, fixes #7709

## Submitter Checklist:


## Test Plan:
Note: Stripe is only supported in the US. If outside of the US, you've have to update device settings to United States region.
1. Open Buy Crypto View
2. Verify `0` displayed as placeholder, `$`  / USD selected as default currency.
3. Enter an amount in the buy field (optional) and tap `Select purchase method`.
4. Verify Stripe is shown for supported tokens. ETH (ethereum mainnet), MATIC (polygon), SOL (solana mainnet), USDC (ethereum mainnet) tokens are supported by Stripe.
5. Tap `Buy with Link`.
6. Verify wallet dismisses and opens the Stripe purchase page

## Screenshots:

https://github.com/brave/brave-ios/assets/5314553/cead20ec-82e2-48b3-a9c8-e4a44cb14e39


## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
